### DISCUSSION TITLE
Hero Transitions (aka `matchedGeometryEffect`)

### DISCUSSION THREAD

**Discussion by davdroman** [davdroman/swiftui-navigation-transitions#4] [Ideas]
I believe there's real potential for the library to introduce [Hero Transitions](https://github.com/HeroTransitions/Hero) for push and pop.

This would be made possible by allowing access to the subviews of the origin and destination [`TransientView`](https://github.com/davdroman/swiftui-navigation-transitions/blob/main/Documentation/Custom-Transitions.md#intermediate)s.

`TransientView` in itself is only a wrapper around the UIView being animated, so one could easily expose further "child" `TransientView`s by tagging them with a SwiftUI modifier to be extracted. This can either be via the existing accessibility identifiers or an entirely new strongly-typed ID type:

```swift
// rough API idea
// fromView is of type TransientView.
fromView.subview(withAccessibilityIdentifier: "logo") // specific subview as TransientView that can now be animated
fromView["logo"] // theoretical shorthand syntax for the above
```

```swift
// identifiers could be made to be strongly typed and specifically applied to views via a modifier 
.navigationTransitionID(.logo) // .logo could be a user-defined enum case conforming to some vended protocol
```

**Comment by Christopher-A** [comment]
@davdroman Have you had any time to think more on this? As SwiftUI matures, designs are getting more complex and a matched geometry effects across navigation transitions are a hotly requested feature.

**Reply by davdroman** [reply]
@Christopher-A before I delve into hero transitions, I need to do some preliminary work to enable custom transitions for more modes of presentation (`.sheet`, `.fullScreenCover`, etc.). Cautiously optimistic to have it all ready by end of summer.

**Reply by davdroman** [reply]
An update on this.

I won't be able to make it for end of summer anymore, as I'm now winding down to become a dad in about a month and my life is about to be quite hectic for a good few months. I'll still carve some time out for general maintenance and bugfixing, but I won't have time to incorporate new features until sometime next year.

I hope this isn't too much of a disappointment, and thank you for your understanding.

**Reply by Christopher-A** [reply]
Congratulations! Fatherhood will be the hardest thing you will ever do. Thankfully it’s also the best!
